### PR TITLE
Enable toggle for params encoding

### LINKED ISSUE #2897: [Feature] Users would like to enable optional encoding of URI for APIs
A default true option for encoding allows users to skip a forced decoding step in case their URI components are already encoded.

### PR DESCRIPTION
**Pull request by sumitsum**
## Description
1. Add toggle param to ActionConfiguration class.
2. Add TCs to test toggle on/off/null.

Fixes #2897 

## Type of change
- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

## How Has This Been Tested?
Tested via Junit TCs. Add 3 TC:
1. for toggle set to true.
2. for toggle set to false.
3. for toggle see to null.

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by trishaanand** [comment]
@sumitsum : If I understood the problem clearly, this was for body of content type form url encoded, NOT the url itself. Could you please confirm the requirements?

File: app/server/appsmith-plugins/restApiPlugin/src/main/java/com/external/plugins/RestApiPlugin.java [hunk 1]
```diff
@@ -285,7 +297,8 @@ public String convertPropertyListToReqBody(List<Property> bodyFormData, String r
                         String key = property.getKey();
                         String value = property.getValue();
 
-                        if (MediaType.APPLICATION_FORM_URLENCODED_VALUE.equals(reqContentType)) {
+                        if (MediaType.APPLICATION_FORM_URLENCODED_VALUE.equals(reqContentType)
+                            && (encodeParamsToggle == null || encodeParamsToggle == true)) {
```
Comment (trishaanand) [inline app/server/appsmith-plugins/restApiPlugin/src/main/java/com/external/plugins/RestApiPlugin.java:301]: Considering that you have initialised `encodeParamsToggle` to true, it would never be null. The extra null check can be removed.

File: app/server/appsmith-plugins/restApiPlugin/src/main/java/com/external/plugins/RestApiPlugin.java [hunk 2]
```diff
@@ -488,16 +501,32 @@ private String addHttpToUrlWhenPrefixNotPresent(String url) {
...
                 for (Property queryParam : queryParams) {
                     String key = queryParam.getKey();
                     if (StringUtils.isNotEmpty(key)) {
-                        uriBuilder.queryParam(URLEncoder.encode(key, StandardCharsets.UTF_8),
-                                URLEncoder.encode(queryParam.getValue(), StandardCharsets.UTF_8));
+                        /*
+                         * - If encodeParamsToggle is null, then assume it to be true because params are supposed to be
+                         *   encoded by default, unless explicitly prohibited by the user.
+                         */
+                        if(encodeParamsToggle == null || encodeParamsToggle == true) {
```
Comment (trishaanand) [inline app/server/appsmith-plugins/restApiPlugin/src/main/java/com/external/plugins/RestApiPlugin.java:518]: Same as above. `encodeParamsToggle` would never be null because its been initialized.

**Comment by trishaanand** [comment]
LGTM. Can be merged.

**Review by trishaanand — APPROVED** [review]

File: app/server/appsmith-plugins/restApiPlugin/src/main/java/com/external/plugins/RestApiPlugin.java (hunk 1 above)
Comment (sumitsum) [reply app/server/appsmith-plugins/restApiPlugin/src/main/java/com/external/plugins/RestApiPlugin.java:301]: Hi @trishaanand I have seen UI forms sending null value sometimes, when user doesn't select / edit the field. This null check will prevent our programs from crashing. e.g please refer to this PR https://github.com/appsmithorg/appsmith/pull/2934

File: app/server/appsmith-plugins/restApiPlugin/src/main/java/com/external/plugins/RestApiPlugin.java (hunk 2 above)
Comment (sumitsum) [reply app/server/appsmith-plugins/restApiPlugin/src/main/java/com/external/plugins/RestApiPlugin.java:518]: Hi @trishaanand I have seen UI forms sending null value sometimes, when user doesn't select / edit the field. This null check will prevent our programs from crashing. e.g please refer to this PR https://github.com/appsmithorg/appsmith/pull/2934

Comment (sumitsum) [reply app/server/appsmith-plugins/restApiPlugin/src/main/java/com/external/plugins/RestApiPlugin.java]: Done.

File: app/server/appsmith-plugins/restApiPlugin/src/main/java/com/external/plugins/RestApiPlugin.java (hunk 1 above)
Comment (sumitsum) [reply app/server/appsmith-plugins/restApiPlugin/src/main/java/com/external/plugins/RestApiPlugin.java]: Done.

**Review by trishaanand — APPROVED** [review]
